accepted/future-releases/constructor-tearoffs/feature-specification.md

```diff
@@ -515,6 +515,24 @@ void main() {
 
 The grammar changes necessary for these changes are provided separately (as [changes to the spec grammar](https://dart-review.googlesource.com/c/sdk/+/197161)). The grammar change examples above are for illustration only.
 
+### Constant expressions
+
+We add the following to the set of expressions that are potentially constant and constant:
+
+If `e` is a potentially constant expression, `T1..Tk` is derived from `<typeList>`, and `e<T1..Tk>` can be derived from `<primary> <selector>*`, then `e<T1..Tk>` is a potentially constant expression.
+If moreover `e` is a constant expression whose static type is a function type `F`, and `T1..Tk` is a list of constant type expressions, then `(e)<T1..Tk>` is a constant expression.
+
+*It follows that `F` is a generic function type taking `k` type arguments, and `T1..Tk` satisfy the bounds of `F`, because otherwise the program would have a compile-time error.*
+
+The following cases are specified elsewhere in this document:
+
+*Named constructor tearoffs:
+Expressions of the form <code>*C*\<*typeArgs*>.*name*</code> can be potentially constant and constant expressions.
+The constantness of named constructor tearoffs follows the constantness of the tearoffs of the corresponding static methods.*
+
+*Tearing off constructors from type aliases:
+Constantness falls in the categories non-generic, generic/instantiated; generic/uninstantiated/proper-rename; generic/uninstantiated/not-proper-rename.*
```

Comment:
Not sure what this says.

Reply:
The intention is that it should just serve as a reminder that the section 'Tearing off constructors from type aliases' contains several rules about constant expressions, organized into 4 different groups. I revised it, and I think it should be considerably more readable now.